Using commons

Lasering opened the pull request.

Lasering committed:
Use scala-openstack-common-clients.

Change name to scala-designate-client.
Update Scala to 2.13.3.
Update dependencies.
Prepare sbt to make releases.
Add readme.
Other minor fixes.

Commit:
Fix ZoneTransferRequest createTransfer method

Lasering committed:
Use the new common snapshot.

Fix some tests.

Lasering committed:
Improve tests.

Setup travis.
Update readme.
Use the latest common clients snapshot.

Lasering committed:
Refactor tests. They are now passing.

Lasering committed:
Use postHandleConflict from the latest commons snapshot.

Lasering commented on src/test/scala/pt/tecnico/dsi/openstack/designate/RecordsetsSpec.scala:
```diff
@@ -0,0 +1,71 @@
+      dummyZone <- designate.zones.create(Zone.Create("recordsets.org.", "john.doe@recordsets.org"))
+      recordsets = designate.zones.recordsets(dummyZone.id)
+      recordsetCreate = Recordset.Create(
+        name = s"sudomain${Random.nextInt()}.recordsets.org.",
```
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
This is in tests, not really a problem.

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+
+Currently supported endpoints:
+
+- [Zones](https://docs.openstack.org/api-ref/dns/#zone)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/test/scala/pt/tecnico/dsi/openstack/designate/ZoneTransferAcceptsSpec.scala:
```diff
@@ -0,0 +1,57 @@
+        keystone <- keystoneClient
+        dummyProject <- keystone.projects.create(Project("dummy", "dummy project", "default"))
+        designate <- client
+        dummyZone <- designate.zones.create(Zone.Create(s"zone${Random.nextInt()}.org.", "joe@example.org"))
```
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+- [Zone Ownership Transfers - Requests](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-requests)
+- [Zone Ownership Transfers - Accepts](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-accepts)
+- [Recordsets](https://docs.openstack.org/api-ref/dns/#recordsets)
+- [Limits](https://docs.openstack.org/api-ref/dns/#limits)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/main/scala/pt/tecnico/dsi/openstack/designate/services/ZoneTransferRequests.scala:
```diff
@@ -0,0 +1,42 @@
+
+  def create(zoneId: String, value: ZoneTransferRequest.Create, extraHeaders: Header*): F[WithId[ZoneTransferRequest]] =
+    super.postHandleConflict(wrappedAt = None, value, createUri(zoneId) / "transfer_requests", extraHeaders:_*) {
+      list(Query.empty, extraHeaders:_*).filter(_.zoneId == zoneId).head.compile.lastOrError.flatMap { existing =>
```
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of head on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/main/scala/pt/tecnico/dsi/openstack/designate/services/FloatingIPs.scala:
```diff
@@ -0,0 +1,26 @@
+  def get(region: String, floatingIpId: String, extraHeaders: Header*): F[WithId[FloatingIP]] =
+    super.get(wrappedAt = None, uri / s"$region:$floatingIpId", extraHeaders:_*)
+
+  def set(region: String, floatingIpId: String, floatingIp: FloatingIP, extraHeaders: Header*): F[WithId[FloatingIP]] =
```
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/main/scala/pt/tecnico/dsi/openstack/designate/services/Recordsets.scala:
```diff
@@ -0,0 +1,26 @@
+final class Recordsets[F[_]: Sync: Client](baseUri: Uri, authToken: Header)
+  extends CrudService[F, Recordset, Recordset.Create, Recordset.Update](baseUri, "recordset", authToken, wrapped = false) {
+
+  def getByName(name: String, extraHeaders: Header*): F[WithId[Recordset]] =
```
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+Currently supported endpoints:
+
+- [Zones](https://docs.openstack.org/api-ref/dns/#zone)
+- [Zone Ownership Transfers - Requests](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-requests)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/main/scala/pt/tecnico/dsi/openstack/designate/services/ZoneTransferAccepts.scala:
```diff
@@ -0,0 +1,46 @@
+    // Once a zone transfer is accepted it is not possible to accept it again. This means this method is not idempotent.
+    // However we can make it idempotent: if an accept for `zoneTransferRequestId` already exists we return it, being careful
+    // to ensure the key is set to the correct value. Otherwise we create the accept.
+    list(Status.Complete, extraHeaders:_*).filter(_.zoneTransferRequestId == zoneTransferRequestId).compile.last.flatMap { optionalAccept =>
```
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Usage of last on collections.](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+- [Zone Ownership Transfers - Accepts](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-accepts)
+- [Recordsets](https://docs.openstack.org/api-ref/dns/#recordsets)
+- [Limits](https://docs.openstack.org/api-ref/dns/#limits)
+- [Quotas](https://docs.openstack.org/api-ref/dns/#quotas)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+- [Zones](https://docs.openstack.org/api-ref/dns/#zone)
+- [Zone Ownership Transfers - Requests](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-requests)
+- [Zone Ownership Transfers - Accepts](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-accepts)
+- [Recordsets](https://docs.openstack.org/api-ref/dns/#recordsets)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+The Scala client for Openstack Designate.
+
+Currently supported endpoints:
+
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on README.md:
```diff
@@ -1 +1,32 @@
+
+- [Zones](https://docs.openstack.org/api-ref/dns/#zone)
+- [Zone Ownership Transfers - Requests](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-requests)
+- [Zone Ownership Transfers - Accepts](https://docs.openstack.org/api-ref/dns/#zone-ownership-transfers-accepts)
```
Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/main/scala/pt/tecnico/dsi/openstack/designate/services/Zones.scala:
```diff
@@ -0,0 +1,42 @@
+final class Zones[F[_]: Sync: Client](baseUri: Uri, authToken: Header)
+  extends CrudService[F, Zone, Zone.Create, Zone.Update](baseUri, "zone", authToken, wrapped = false) { self =>
+
+  def getByName(name: String, extraHeaders: Header*): F[WithId[Zone]] =
```
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

afonsomatos commented on openrc-designate.sh:
```diff
@@ -0,0 +1,8 @@
+export OS_PROJECT_NAME=admin
+export OS_PROJECT_DOMAIN_NAME=default
+export OS_IDENTITY_API_VERSION=3
+export OS_AUTH_URL=http://centos-sentmwlu.example-service-dev.al.vps.tecnico.ulisboa.pt:5000/v3
```
Again, the URL of the VPS.

Lasering replied:
We need to test against something, what we have at our disposal is a VM in our VPS.

This is also related to the discussion below. If we change the VM we are testing against then we might also need to change the default quotas.

afonsomatos replied:
Why don't we include these links in some private repo or something? Sounds weird to opensource these links...

afonsomatos commented on src/test/scala/pt/tecnico/dsi/openstack/designate/QuotasSpec.scala:
```diff
@@ -0,0 +1,66 @@
+        zoneRecordsets = 25,
+        zones = 15
+      )
+      designate.quotas.update(dummyProjectId, newQuotas, allProjectsHeader).idempotently( _ shouldBe newQuotas)
```
This makes a lot of sense. But the previous test, where we check for _our_ defaults doesn't seem right. Afterall, the test will not pass if the defaults have changed.

Lasering replied:
You have no way of knowing the default quotas without asking for them. And if you did that the test would be useless.

Since this is a client we are not really interested in the real value of the quotas/default quotas but rather that the client can successfully parse them.

afonsomatos replied:
What about an update followed by a get, instead of just a get? This way the test would always pass inrespective of the defaults, since you would be updating it first. At the end of the day, the integration test would continue to test update and get, which is what we want.

Lasering commented on src/test/scala/pt/tecnico/dsi/openstack/designate/RecordsetsSpec.scala:
```diff
@@ -0,0 +1,71 @@
+      dummyZone <- designate.zones.create(Zone.Create("recordsets.org.", "john.doe@recordsets.org"))
+      recordsets = designate.zones.recordsets(dummyZone.id)
+      recordsetCreate = Recordset.Create(
+        name = s"sudomain${Random.nextInt()}.recordsets.org.",
```
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering committed:
Fix travis wrong before_script.

Lasering committed:
Force Coursier TTL to a low value.

Lasering committed:
Disable coverage.

Lasering committed:
IPv6 tests

Lasering committed:
Major refactor to tests. They now use Resource.

Use the new test VM.
Force Java to resolve names to IPv6 first.
Small improvements to build.sbt

Lasering commented on src/main/scala/pt/tecnico/dsi/openstack/designate/services/FloatingIPs.scala:
```diff
@@ -0,0 +1,28 @@
+  def get(region: String, floatingIpId: String, extraHeaders: Header*): F[WithId[FloatingIP]] =
+    super.get(wrappedAt = None, uri / s"$region:$floatingIpId", extraHeaders:_*)
+
+  def set(region: String, floatingIpId: String, floatingIp: FloatingIP.Create, extraHeaders: Header*)
```
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Use active names for operations with side effects](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/test/scala/pt/tecnico/dsi/openstack/designate/Utils.scala:
```diff
@@ -32,23 +35,38 @@ abstract class Utils extends AsyncWordSpec with Matchers with BeforeAndAfterAll
-    designateUrl <- IO.fromEither(designateUrlOpt.toRight(new Throwable("Could not find \"dns\" service in the catalog")))
-  } yield new DesignateClient[IO](Uri.unsafeFromString(designateUrl), keystone.authToken)
+  // Not very purely functional :(
+  val random = new Random()
```
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering commented on src/test/scala/pt/tecnico/dsi/openstack/designate/Utils.scala:
```diff
@@ -32,23 +35,38 @@ abstract class Utils extends AsyncWordSpec with Matchers with BeforeAndAfterAll
-    designateUrl <- IO.fromEither(designateUrlOpt.toRight(new Throwable("Could not find \"dns\" service in the catalog")))
-  } yield new DesignateClient[IO](Uri.unsafeFromString(designateUrl), keystone.authToken)
+  // Not very purely functional :(
+  val random = new Random()
```
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)

Lasering replied:
Codacy found an issue: [Don't use a predictable random number generator](https://app.codacy.com/gh/ist-dsi/scala-designate-client/pullRequest?prid=5821613)
